companion/src/modeledit/logicalswitches.cpp

```diff
@@ -303,52 +333,56 @@ void LogicalSwitchesPanel::updateTimerParam(QDoubleSpinBox *sb, int timer, doubl
 #define VALUE_TO_VISIBLE 0x20
 #define DELAY_ENABLED    0x40
 #define DURATION_ENABLED 0x80
+#define LINE_ENABLED     0x100
 
-void LogicalSwitchesPanel::setSwitchWidgetVisibility(int i)
+void LogicalSwitchesPanel::updateLine(int i)
 {
   lock = true;
   unsigned int mask;
 
+  csw[i]->setCurrentIndex(csw[i]->findData(model->logicalSw[i].func));
+  cswitchAnd[i]->setCurrentIndex(cswitchAnd[i]->findData(RawSwitch(model->logicalSw[i].andsw).toValue()));
+
   if (!model->logicalSw[i].func) {
     mask = 0;
   }
   else {
-    mask = DELAY_ENABLED | DURATION_ENABLED;
+    mask = LINE_ENABLED | DELAY_ENABLED | DURATION_ENABLED;
+
     switch (model->logicalSw[i].getFunctionFamily())
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a "default" case to this switch statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)